akka-typed/src/main/scala/akka/typed/cluster/ddata/scaladsl/Replicator.scala

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster.ddata.scaladsl
+
+import akka.actor.NoSerializationVerificationNeeded
+
+import akka.cluster.{ ddata ⇒ dd }
+import akka.cluster.ddata.Key
+import akka.cluster.ddata.ReplicatedData
+import akka.typed.ActorRef
+import akka.typed.Behavior
+import akka.typed.cluster.ddata.internal.ReplicatorBehavior
+
+object Replicator {
+
+  def behavior(settings: ReplicatorSettings): Behavior[Command[_]] =
+    ReplicatorBehavior.behavior(settings)
+
+  type ReadConsistency = dd.Replicator.ReadConsistency
+  val ReadLocal = dd.Replicator.ReadLocal
+  type ReadFrom = dd.Replicator.ReadFrom
+  type ReadMajority = dd.Replicator.ReadMajority
+  type ReadAll = dd.Replicator.ReadAll
+
+  type WriteConsistency = dd.Replicator.WriteConsistency
+  val WriteLocal = dd.Replicator.WriteLocal
+  type WriteTo = dd.Replicator.WriteTo
+  type WriteMajority = dd.Replicator.WriteMajority
+  type WriteAll = dd.Replicator.WriteAll
+
+  sealed trait Command[A <: ReplicatedData] {
+    def key: Key[A]
+  }
+
+  final case class Get[A <: ReplicatedData](key: Key[A], consistency: ReadConsistency, request: Option[Any] = None)(val replyTo: ActorRef[GetResponse[A]])
+    extends Command[A] {
+
+  }
+
+  type GetResponse[A <: ReplicatedData] = dd.Replicator.GetResponse[A]
+  object GetSuccess {
+    def unapply[A <: ReplicatedData](rsp: GetSuccess[A]): Option[(Key[A], Option[Any])] = Some((rsp.key, rsp.request))
+  }
+  type GetSuccess[A <: ReplicatedData] = dd.Replicator.GetSuccess[A]
+  type NotFound[A <: ReplicatedData] = dd.Replicator.NotFound[A]
+  type GetFailure[A <: ReplicatedData] = dd.Replicator.GetFailure[A]
```

Comment:
My thinking is that we could use the same messages as untyped.Replicator for response messages, but we have to use different for the requests, since we have to map the `sender`.

Replies can then be sent directly back to the `replyTo` actor.